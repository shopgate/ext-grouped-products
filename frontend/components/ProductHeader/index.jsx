import React from 'react';
import PropTypes from 'prop-types';
import Rating from './components/Rating';
import connect from './connector';
import styles from './style';

/**
 * The ProductHeader component. It's supposed to hide the default header
 * when the current active product has grouped products and render necessary elements.
 * @param {Object} props The component props.
 * @param {boolean} props.isRendered Should the component be rendered.
 * @return {JSX}
 */
const ProductHeader = ({ isRendered }) => {
  if (!isRendered) {
    return null;
  }

  return (
    <div className={styles.content}>
      <Rating />
    </div>
  );
};

ProductHeader.propTypes = {
  isRendered: PropTypes.bool.isRequired,
};

export default connect(ProductHeader);
