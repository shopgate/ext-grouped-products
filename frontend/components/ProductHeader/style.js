import { css } from 'glamor';
import { themeConfig } from '@shopgate/pwa-common/helpers/config';

const { variables, colors } = themeConfig;

const content = css({
  ' > div': {
    // Apply styling to the RatingStars component to render the ProductHeader nicely.
    padding: variables.gap.big,
    marginBottom: 0,
    backgroundColor: colors.light,
    borderTop: 'rgba(0, 0, 0, 0.05) 2px solid',
  },
  ' + div': {
    // Hide the original ProductHeader.
    display: 'none',
  },
});

export default {
  content,
};
