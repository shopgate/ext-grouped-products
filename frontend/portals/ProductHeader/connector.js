import { connect } from 'react-redux';
import { hasGroupedProducts } from '../../selectors';

/**
 * Maps the contents of the state to the component props.
 * @param {Object} state The current application state.
 * @param {Object} props Props.
 * @return {Object} The extended component props.
 */
const mapStateToProps = (state, props) => ({
  isRendered: hasGroupedProducts(state, props),
});

export default connect(mapStateToProps);
