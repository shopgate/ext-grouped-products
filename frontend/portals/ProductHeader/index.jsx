import { withPageProductId } from '@shopgate-ps/pwa-extension-kit/connectors';
import { connect } from 'react-redux';
import { hasGroupedProducts } from "../../selectors";

const ProductHeader = ({ children, hasGroupedProducts }) => {
    if (hasGroupedProducts) {
        return null;
    }

    return children;
}

const mapStateToProps = (state, props) => ({
    hasGroupedProducts: hasGroupedProducts(state, props)
});

const ConnectedProductHeader = connect(mapStateToProps)(ProductHeader);

export default withPageProductId(ConnectedProductHeader);
