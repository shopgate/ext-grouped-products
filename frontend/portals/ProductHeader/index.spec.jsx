import React from 'react';
import { mount } from 'enzyme';
import ProductHeader from './index';

jest.mock('@shopgate-ps/pwa-extension-kit/connectors', () => ({
  withPageProductId: Component => props => <Component {...props} />,
}));

let mockedHasGroupedProducts = false;
jest.mock('react-redux', () => ({
  connect: () => Component => props =>
    <Component hasGroupedProducts={mockedHasGroupedProducts} {...props} />,
}));

describe('ProductHeader', () => {
  const Children = () => <div>Hello world</div>;
  it('should return null when product has grouped products', () => {
    mockedHasGroupedProducts = true;
    const component = mount(<ProductHeader><Children /></ProductHeader>);
    expect(component.html()).toBe(null);
  });

  it('should return original header when product has no grouped products', () => {
    mockedHasGroupedProducts = false;
    const component = mount(<ProductHeader><Children /></ProductHeader>);
    expect(component.find('div').exists()).toBe(true);
    expect(component).toMatchSnapshot();
  });
});
