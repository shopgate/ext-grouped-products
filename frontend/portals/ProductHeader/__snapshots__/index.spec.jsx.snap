// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductHeader should return original header when product has no grouped products 1`] = `
<Component>
  <Component>
    <ProductHeader
      hasGroupedProducts={false}
    >
      <Children>
        <div>
          Hello world
        </div>
      </Children>
    </ProductHeader>
  </Component>
</Component>
`;
